### PR TITLE
Add a new check for array.prototype.sort in array prototype dispatch

### PR DESCRIPTION
**Pull request by kisbg**
Bugfix: first check arguments callable then the length of the this value.

JerryScript-DCO-1.0-Signed-off-by: bence gabor kis kisbg@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c [hunk 1]
```diff
@@ -2958,6 +2952,15 @@ ecma_builtin_array_prototype_dispatch_routine (uint16_t builtin_routine_id, /**<
   }
 #endif /* ENABLED (JERRY_ESNEXT) */
 
+  /* Check if the provided compare function is callable. */
+  if (builtin_routine_id == ECMA_ARRAY_PROTOTYPE_SORT
+      && !ecma_is_value_undefined (arguments_list_p[0])
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c:2957]: Please rearrange this builtin id to not burden the common code path.

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c [hunk 2]
```diff
@@ -1108,15 +1108,23 @@ ecma_builtin_array_prototype_object_sort_compare_helper (ecma_value_t lhs, /**<
...
   }
 
+  ecma_length_t len;
+  ecma_value_t len_value = ecma_op_object_get_length (obj_p, &len);
+
+  if (ECMA_IS_VALUE_ERROR (len_value))
+  {
+    return len_value;
+  }
+  ecma_free_value (len_value);
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c:1126]: This should not be released.

**Review by rerobika — CHANGES_REQUESTED** [review]

**Review by zherczeg — APPROVED** [review]
LGTM

**Review by rerobika — APPROVED** [review]
LGTM

**Pull request merged**